piksi_tools/fileio.py

```diff
@@ -351,12 +559,13 @@ def write(self, filename, data, offset=0, trunc=True, progress_cb=None):
                 msg = MsgFileioWriteReq(
                     sequence=seq,
                     offset=offset,
-                    filename=filename + b'\x00',
-                    data=chunk)
+                    filename=filename + b'\x00' + chunk,
```

Comment:
Why like this?

Reply:
This one deserves an entire post-mortem on it own.

Reply:
I'll add a comment, but basically it's 2 or 3 times faster (FileIO itself) if we use the "wrong" form of this.

Reply:
I added a comment about this, but also experimented with some other options, some of which resulted in pretty large speed-ups (30% increase in max transfer speed) such as this one: https://github.com/swift-nav/piksi_tools/commit/0ebf8d91d7870c30e5a51c142427b58f985b313f -- which basically adds numba (JIT compiler for individual Python functions) to speed up the CRC calculation and also hand rolled the write message packet.  There's a lot of other work that could probably be done too, like trying to avoid copies in that pipeline, but wasn't able to get that work yesterday.